prowler/providers/aws/services/apigateway/apigateway_restapi_authorizers_enabled/apigateway_restapi_authorizers_enabled.py

```diff
@@ -14,27 +14,60 @@ def execute(self):
+                report.status_extended = f"API Gateway {rest_api.name} ({rest_api.id}) has an authorizer configured at the API level"
             else:
                 # we want to know if api has not authorizers and all the resources don't have methods configured
-                api_resources_with_all_methods_authorized = False
+                paths_without_authorizers = {}
                 for resource in rest_api.resources:
                     # if the resource has methods test if they have all configured authorizer
                     if resource.resource_methods:
-                        if "NONE" not in resource.resource_methods.values():
-                            api_resources_with_all_methods_authorized = True
-                        else:
-                            # with only one unauthorized method -> return FAIL
-                            api_resources_with_all_methods_authorized = False
-                            break
+                        unauthorized_methods = [
+                            method
+                            for method, authorizer in resource.resource_methods.items()
+                            if authorizer == "NONE" and not method == "OPTIONS"
+                        ]
+                        if unauthorized_methods:
+                            paths_without_authorizers[
+                                resource.path
+                            ] = unauthorized_methods
 
-                if api_resources_with_all_methods_authorized:
+                if paths_without_authorizers:
+                    report.status = "FAIL"
+                    failed_status_parts = []
+                    for path, methods in paths_without_authorizers.items():
+                        methods_str = self.array_to_string(methods)
+                        if len(methods) == 1:
+                            failed_status_parts.append(
+                                f"{path} without authorizer for the {methods_str} method"
+                            )
+                        else:
+                            failed_status_parts.append(
+                                f"{path} without authorizers for {methods_str} methods"
+                            )
+                    failed_status_string = "; ".join(failed_status_parts)
+                    report.status_extended = f"API Gateway {rest_api.name} ({rest_api.id}) issues: {failed_status_string}"
+                else:
                     report.status = "PASS"
-                    report.status_extended = f"API Gateway {rest_api.name} ID {rest_api.id} has an authorizer configured at api or methods level."
+                    report.status_extended = f"API Gateway {rest_api.name} ({rest_api.id}) has all methods authorized."
 
             findings.append(report)
 
         return findings
+
+    @staticmethod
+    def array_to_string(arr):
```

Comment:
I think there is no need for this function, it is enough if you show the values comma-separated, like:
```python
>>> ", ".join(['A', 'B', 'C'])
'A, B, C'
```

Reply:
I have removed it